controller/src/controller/models/domain/rsptag.py

```diff
@@ -349,6 +368,55 @@ def _determine_build(
         else:
             return rest if rest else None
 
+    @staticmethod
+    def _calculate_date(tagdata: dict[str, str]) -> datetime.datetime | None:
+        """Calculate the date when the image should have been created.
+
+        Parameters
+        ----------
+        tagdata
+            The match groups from the regular expression tag match.
+
+        Returns
+        -------
+        datetime.datetime | None
+            The image creation date if it can be gleaned from the tag.
+        """
+        year = tagdata.get("year")
+        if not year:
+            return None
+        week = tagdata.get("week")
+        if year and week:
+            thursday = 4  # We build on Thursday, and Monday is index 1.
+            stamp = datetime.datetime.fromisocalendar(
+                int(year), int(week), thursday
+            )
+            # fromisocalendar() gives us a naive object; force it to UTC
+            return stamp.replace(tzinfo=datetime.UTC)
```

Comment:
```suggestion
            # Weekly images are built on Thursday (ISO weekday 4).
            stamp = datetime.fromisocalendar(int(year), int(week), 4)
            return stamp.replace(tzinfo=UTC)
```
I don't think commenting the `replace` call is needed since its existence communicates the same information as the comment.